Source/Task/AsyncLib.cpp

```diff
@@ -179,6 +279,9 @@ static HRESULT AllocStateNoCompletion(_In_ AsyncBlock* asyncBlock)
 #endif
     }
 
+    // construction is inherently single threaded
+    // (there is nothing we can do if the client tries to use the same
+    // AsyncBlock in 2 calls at the same time)
     new (asyncBlock->internal) AsyncBlockInternal{ state.Detach() };
```

Comment:
I'd move this to the very beginning so we can set status on the block always with no error path.